x-pack/plugin/runtime-fields/qa/rest/src/yamlRestTest/java/org/elasticsearch/xpack/runtimefields/rest/CoreTestsWithRuntimeFieldsIT.java

```diff
@@ -173,4 +195,54 @@ private static String painlessToLoadFromSource(String name, String type) {
         )
     );
 
+    private static final ExecutableSection ADD_TEMPLATE = new ExecutableSection() {
+        @Override
+        public XContentLocation getLocation() {
+            return new XContentLocation(-1, -1);
+        }
+
+        @Override
+        public void execute(ClientYamlTestExecutionContext executionContext) throws IOException {
+            Map<String, String> params = Map.of("name", "convert_to_source_only", "create", "true");
+            List<Map<String, Object>> dynamicTemplates = new ArrayList<>();
+            for (String type : PAINLESS_TO_EMIT.keySet()) {
+                if (type.equals("ip")) {
+                    // There isn't a dynamic template to pick up ips. They'll just look like strings.
+                    continue;
+                }
+                Map<String, Object> mapping = Map.ofEntries(
+                    Map.entry("type", "runtime_script"),
+                    Map.entry("runtime_type", type),
+                    Map.entry("script", painlessToLoadFromSource("{name}", type))
+                );
+                Map<String, Object> body = Map.ofEntries(
+                    Map.entry("match_mapping_type", type.equals("keyword") ? "string" : type),
```

Comment:
what is `string` here? Is it the old field type `string` from before we had `text` and `keyword`?

Reply:
I think it is! Dynamic templates mostly uses the types that we use but it uses `string` instead of `text` or `keyword`.